src/main/java/ch/heigvd/dai/commands/Hide.java

```diff
@@ -0,0 +1,85 @@
+package ch.heigvd.dai.commands;
+
+import ch.heigvd.dai.utilities.BmpFile;
+import ch.heigvd.dai.utilities.FileManipulator;
+import ch.heigvd.dai.utilities.MessageManipulator;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.concurrent.Callable;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "hide",
+    description = "Hide the bits of a message/file into a BMP image.")
+public class Hide implements Callable<Integer> {
+  @CommandLine.ParentCommand private Root parent;
+
+  @CommandLine.Parameters(
+      index = "0",
+      description = "The filename where the image with the hidden content will be stored.",
+      defaultValue = "output.bmp")
+  private String filenameOutput;
+
+  @CommandLine.Option(
+      names = {"-b", "--bits-per-byte"},
+      description =
+          "Number of bits to hide per byte of information of the BMP image. Note that the original bits of the image will be overwritten and it may be more noticeable that something is hidden within the image.",
+      defaultValue = "1")
+  private int bitsPerByte;
+
+  @Override
+  public Integer call() {
+    if (!isPowerOfTwo(bitsPerByte) || bitsPerByte < 1 || bitsPerByte > 8) {
+      System.err.println("--bits-per-byte needs to be a power of 2 between 1 and 8 inclusively");
+      return 1;
+    }
+
+    if (parent.forceDisabled() && Files.exists(Paths.get(filenameOutput))) {
```

Comment:
I'm not sure we're allowed to use these, since they come from `java.nio`? Open to your opinion